docs/src/philosophy.md

```diff
@@ -0,0 +1,127 @@
+# Philosophy
+
+The goal of the `Interval` type is to be directly used to replace floating point
+number in arbitrary julia code, such that in any calculation,
+the resulting intervals are guaranteed to bound the true image of the starting
+intervals.
+
+Due to the way that the julia ecosystem has evolved,
+this means that `Interval` **must** be a subtype of `Real`,
+as it is the default supertype used to describe
+"numbers that are not complex".
```

Comment:
Maybe "must" is too strong?

"
Due to the evolution of the Julia ecosystem -- specifically that `Real` serves as the default supertype for numerical types that are not complex --, it is particularly advantageous for `Interval` to be defined as a subtype of `Real`.
"

Reply:
I have tuned it down a little bit, but I like "must" so that it is clear it is not open for discussion (at least for the current document).